spec/iso8601/time_interval_spec.rb

```diff
@@ -0,0 +1,375 @@
+      expect(ISO8601::TimeInterval.new(duration_with_base).to_s).to eq('2010-05-09T10:30:12+00:00/P1Y1M1DT0.5H')
+    end
+  end
+
+  describe "compare Time Intervals" do
+    before(:each) do
+      @small = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+      @big = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT2H')
+    end
+
+    it "should raise TypeError when compared object is not a ISO8601::TimeInterval or Numeric" do
+      expect { @small < 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+      expect { @small > 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+      expect { @small == 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+    end
+
+    it "should check what interval is bigger" do
+      expect(@small > @big).to be_falsy
+      expect(@big > @small).to be_truthy
+      expect(@small > @small).to be_falsy
+      expect(@small > @small.to_f).to be_falsy
+      expect(@small > (@small.to_f + 60 * 60)).to be_falsy
+      expect(@small > (@small.to_f - 60 * 60)).to be_truthy
+    end
+
+    it "should check if interval is bigger or equal than other" do
+      expect(@small >= @big).to be_falsy
+      expect(@big >= @small).to be_truthy
+      expect(@small >= @small).to be_truthy
+      expect(@small >= @small.to_f).to be_truthy
+      expect(@small >= (@small.to_f + 60 * 60)).to be_falsy
+      expect(@small >= (@small.to_f - 60 * 60)).to be_truthy
+    end
+
+    it "should check what interval is smaller" do
+      expect(@small < @big).to be_truthy
+      expect(@big < @small).to be_falsy
+      expect(@small < @small).to be_falsy
+      expect(@small < @small.to_f).to be_falsy
+      expect(@small < (@small.to_f + 60 * 60)).to be_truthy
+      expect(@small < (@small.to_f - 60 * 60)).to be_falsy
+    end
+
+    it "should check if interval is smaller or equal than other" do
+      expect(@small <= @big).to be_truthy
+      expect(@big <= @small).to be_falsy
+      expect(@small <= (@small.to_f + 60 * 60)).to be_truthy
+      expect(@small <= (@small.to_f - 60 * 60)).to be_falsy
+      expect(@small <= @small).to be_truthy
+      expect(@small <= @small.to_f).to be_truthy
+    end
+
+    it "should check if the intervals are equals" do
+      expect(@small == @small).to be_truthy
+      expect(@small == @small.to_f).to be_truthy
+      expect(@small == @big).to be_falsy
+      expect(@small == @big.to_f).to be_falsy
+    end
+  end
+
+  describe "#eql?" do
+    it "should be equal only when start_time and end_time are the same" do
+      interval = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+      interval2 = ISO8601::TimeInterval.new('2007-03-01T14:00:00Z/PT1H')
+      interval3 = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+
+      expect(interval.eql?(interval2)).to be_falsy
+      expect(interval.eql?(interval3)).to be_truthy
+    end
+  end
+
+  describe "#include?" do
```

Comment:
But `Duration` must include a `base` or I can't determine if it's included in the interval